x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -191,15 +191,25 @@ export const reindexServiceFactory = (
 
     const { settings, mappings } = transformFlatSettings(flatSettings);
 
-    const { body: createIndex } = await esClient.indices.create({
-      index: newIndexName,
-      body: {
-        settings,
-        mappings,
-      },
-    });
+    let createIndex;
+    try {
+      createIndex = await esClient.indices.create({
+        index: newIndexName,
+        body: {
+          settings,
+          mappings,
+        },
+      });
+    } catch (err) {
+      // If for any reason the new index name generated by the `generateNewIndexName` already
+      // exists (this could happen if kibana is restarted during reindexing), we can just go
+      // ahead with the process without needing to create the index again.
+      if (err?.body?.error?.type !== 'resource_already_exists_exception') {
+        throw err;
```

Comment:
I think here we want to throw the error returned by ES

Reply:
I feel it might be better to keep the actual es error in here, because if something fails the error it returns will give us a better understanding of why it failed. wdyt?